test/transports/transport-manager.node.js

```diff
@@ -40,10 +50,16 @@ describe('Transport Manager (TCP)', () => {
   })
 
   it('should be able to listen', async () => {
+    sinon.spy(tm, '_createSelfPeerRecord')
+
     tm.add(Transport.prototype[Symbol.toStringTag], Transport)
     await tm.listen(addrs)
     expect(tm._listeners).to.have.key(Transport.prototype[Symbol.toStringTag])
     expect(tm._listeners.get(Transport.prototype[Symbol.toStringTag])).to.have.length(addrs.length)
+
+    // Created Self Peer record on new listen address
+    expect(tm._createSelfPeerRecord.callCount).to.equal(addrs.length)
```

Comment:
Might be better to validate the record is being stored in the address book. Validating the correct record is being stored would also be useful.